oodeel/models/training_funs_torch/torch_models.py

```diff
@@ -155,31 +170,32 @@ def _augment_fn(images, labels):
 
 
 def _train(
-    model,
-    train_data,
-    validation_data,
-    epochs,
-    criterion,
-    optimizer,
-    lr_scheduler,
-    save_dir,
-    device,
+    model: nn.Module,
+    train_data: DataLoader,
+    epochs: int,
+    optimizer: torch.optim.Optimizer,
+    criterion: torch.nn.modules.loss._Loss,
+    device: torch.device,
+    lr_scheduler: torch.optim.lr_scheduler._LRScheduler = None,
+    save_dir: str = None,
+    validation_data: DataLoader = None,
 ):
-    """Torch training loop over tfds dataset
+    """Torch basic training loop
 
     Args:
-        model (_type_): _description_
-        train_data (_type_): _description_
-        validation_data (_type_): _description_
-        epochs (_type_): _description_
-        criterion (_type_): _description_
-        optimizer (_type_): _description_
-        lr_scheduler (_type_): _description_
-        save_dir (_type_): _description_
-        device (_type_): _description_
+        model (nn.Module): Model to train.
+        train_data (DataLoader): Train dataloader.
+        epochs (int): Number of training epochs.
+        optimizer (torch.optim.Optimizer): Optimizer.
+        lr_scheduler (torch.optim.lr_scheduler._LRScheduler): Learning rate scheduler.
+            Defaults to None.
+        criterion (torch.nn.modules.loss._Loss): Criterion for loss.
+        device (torch.device): On which device to train (CUDA or CPU).
```

Comment:
I don't know if it is important for documentation but criterion and device docstrings are after lr_scheduler, whereas it's the opposite in the function's signature.